src/coordinate.rs

```diff
@@ -188,83 +266,196 @@ impl PartialEq for Col {
+            row_cols: fragments,
+        }
+    }};
 }
 
 #[macro_export]
 macro_rules! coord_col {
-    ( $parent_str:tt, $col_str:tt ) => {
-        {
-            let mut col: u32 = 0;
-            for ch in $col_str.to_string().chars() {
-                col += (ch as u32) - 64;
-            }
-
-            Col(coord!($parent_str), NonZeroU32::new(col).unwrap())
+    ( $parent_str:tt, $col_str:tt ) => {{
+        let mut col: u32 = 0;
+        for ch in $col_str.to_string().chars() {
+            col += (ch as u32) - 64;
         }
-    };
+
+        Col(coord!($parent_str), NonZeroU32::new(col).unwrap())
+    }};
 }
 
 #[macro_export]
 macro_rules! coord_row {
-    ( $parent_str:tt, $row_str:tt ) => {
-        {
-            let row: u32 = $row_str.parse::<u32>().unwrap();
+    ( $parent_str:tt, $row_str:tt ) => {{
+        let row: u32 = $row_str.parse::<u32>().unwrap();
 
-            Row(coord!($parent_str), NonZeroU32::new(row).unwrap())
-        }
-    };
+        Row(coord!($parent_str), NonZeroU32::new(row).unwrap())
+    }};
+}
+
+#[cfg(test)]
+mod tests {
+    // Note this useful idiom: importing names from outer (for mod tests) scope.
+    use super::*;
+
+    #[test]
+    fn test_row() {
+        assert_eq!(coord!("root-A1-B2-B3").row().get(), 3);
+    }
+
+    #[test]
+    // - parent.row_cols.len() == result.row_cols.len() - 1
+    fn test_childOf() {
+        // let param = coord!().row_cols.len();
+        // assert_eq!(
+        //     coord!("root-A1-B2-B3").child_of().len(),
+        //     coord!("root-A1-B2-B3").len() + 1
+        // )
+        unimplemented!();
+    }
+
+    #[test]
+    fn test_parent() {
+        assert_eq!(coord!("root").parent(), None);
+        assert_eq!(coord!("meta").parent(), None);
+        // assert_eq!(coord!().parent().len(), coord!().parent().len() + 1); //Check with korede for parent
+
+        // unimplemented!();
+    }
+
+    #[test]
+    fn test_to_string() {
+        assert_eq!(coord!("root-A1-B2-B3").to_string(), "root-A1-B2-B3");
+    }
+
+    #[test]
+    fn test_row_mut() {
+        // assert_eq!(coord!("root-A1-B2-B3").row().get(), 3); // Test for mutability required
+        unimplemented!()
+    }
+
+    #[test]
+    fn test_full_row() {
+        assert_ne!(
+            coord!("root-A1-B2-B3").full_row(),
+            coord_row!("root-A1-B1", "3")
+        );
+        assert_eq!(
+            coord!("root-A1-B2-B3").full_row(),
+            coord_row!("root-A1-B2", "3")
+        );
+    }
+
+    #[test]
+    fn test_row_to_string() {
+        // - row_to_string(coord!("root-A1-B2-B3")) = "root-A1-B2-3"
+        assert_eq!(coord!("root-A1-B2-B3").row_to_string(), "root-A1-B2-3");
+        // info!("{:?}", coord!("root").row_to_string())
+        assert_eq!(coord!("root").row_to_string(), "1");
+        assert_eq!(coord!("meta").row_to_string(), "1");
+        // unimplemented!();
+    }
+
+    #[test]
+    fn test_is_n_parent() {
+        unimplemented!();
```

Comment:
here's how you could test this
```suggestion
        assert_eq!(coord!("root-A1-B2").is_n_parent(coord!("root-A1-B2-B3")), Some(1))
        assert_eq!(coord!("root-A1").is_n_parent(coord!("root-A1-B2-B3")), Some(2))
        assert_eq!(coord!("root-A1-C3").is_n_parent(coord!("root-A1-B2-B3")), None)
        assert_eq!(coord!("root-A1-B2-B4").is_n_parent(coord!("root-A1-B2-B3")), Some(0))
```

note that, if these tests fail, then there must be some problem with my implementation because i haven't had to verify it yet.